jpa/subsystem/src/main/java/org/jboss/as/jpa/puparser/PersistenceUnitXmlParser.java

```diff
@@ -94,8 +94,10 @@ public static PersistenceUnitMetadataHolder parse(final XMLStreamReader reader,
                 version = Version.JPA_2_2;
             } else if ("2".equals(versionString)) {
                 version = Version.JPA_2_0;
+            } else if ("3.0".equals(versionString)) {
+                version = Version.JPA_3_0;
             } else {
-                version = Version.JPA_2_1;
+                version = Version.JPA_2_1; // TODO is this still the right default?
```

Comment:
Good question, for EE 8 I would like to default to the equivalent Jakarta (EE 8) Persistence 2.2 (http://xmlns.jcp.org/xml/ns/persistence/persistence_2_2.xsd) as parsePU() is doing.

For EE 9, I would like to default to 3.0 as you did below for parsePU().

Reply:
@scottmarlow I added a static initializer to see if an EE 9 api class is available and determine the default based on that.

As part of doing that I also centralized the lookup logic and the version # strings a bit.

Reply:
@scottmarlow Does the change I describe ^^^ look good?